### DISCUSSION TITLE
Does Eclipse really have an editor available for Sleigh files in its marketplace or is this just a misleading message?

### DISCUSSION THREAD

**Discussion by hippietrail** [NationalSecurityAgency/ghidra#7114] [Q&A] [answered]
I don't use Eclipse other than for Ghidra. When I open a Sleigh file such as a `.cspec` or `.slaspec` file in Eclipse I get a dialog like this: 
<img width="596" alt="image" src="https://github.com/user-attachments/assets/ebc10eb5-24f1-4b39-ab48-9b8080ff9d3b">
 > Your '*.scpec' file was opened in a simple text editor. Better editor support is available on the Marketplace.
> Show IDE extensions for this file type and let me install them.

But when I click on OK and go to the marketplace I only seem to see totally unrelated extensions:
<img width="593" alt="image" src="https://github.com/user-attachments/assets/17e3b1a3-65f0-4d5c-898a-c18959dd82de">

Am I missing something or is this just a dumb nag dialog that knows nothing about the file type or file extension that brought me here and only shows ads for all extensions?

Are there really extensions there for editing Sleigh files and I just have to jump through some hoops to get to them?

**Comment by ryanmkurtz** [comment, marked as answer]
That is just a dumb generic nag dialog from Eclipse.

However, we do ship the [GhidraSleighEditor](https://github.com/NationalSecurityAgency/ghidra/tree/master/GhidraBuild/EclipsePlugins/GhidraSleighEditor/ghidra.xtext.sleigh) Eclipse plugin with our releases.
